actions/commit-and-push/action.yaml

```
@@ -39,10 +43,16 @@ runs:
       if: ${{ inputs.add-untracked == 'true' }}
       shell: bash
 
-    - name: Commit files
+    - name: Commit files, exit 1 if nothing to commit
+      if: "! inputs.pass-empty-commit"
       run: git commit -m "${{ inputs.commit-message }}" -a
       shell: bash
 
+    - name: Commit files, exit 0 if nothing to commit and no uncommitted changes
+      if: "inputs.pass-empty-commit"
+      run: '( [ -z "$(git status --porcelain=v1 2>/dev/null)"] && echo "No uncommitted changes." ) || git commit -m "${{ inputs.commit-message }}" -a'
```

Bash doesn't normally like `[` and `]` without spaces. Without it, it throws a warning locally for me:
```suggestion
      run: '( [ -z "$(git status --porcelain=v1 2>/dev/null)" ] && echo "No uncommitted changes." ) || git commit -m "${{ inputs.commit-message }}" -a'
```